src/compute/src/logging/differential.rs

```diff
@@ -305,6 +282,17 @@ impl DemuxHandler<'_, '_, '_> {
         let diff = Diff::cast_from(event.diff);
         debug_assert_ne!(diff, 0);
         self.output.sharing.give(self.cap, ((op, ()), ts, diff));
+
+        if let Some(logger) = &mut self.shared_state.compute_logger {
+            let sharing = self.state.sharing.entry(op).or_default();
+            *sharing = (i64::try_from(*sharing).expect("must fit") + diff)
+                .try_into()
+                .expect("under/overflow");
+            if *sharing == 0 {
+                self.state.sharing.remove(&op);
+                logger.log(ComputeEvent::ArrangementHeapSizeOperatorDrop { operator: op });
+            }
+        }
```

Comment:
This works because `compute_logger` will always be `Some` in practice, but if we ever change the code to make it temporarily `None` the `sharing` tracking might become inconsistent. How about we always update `sharing` and only gate the actual `log` call behind `if let Some(logger)`?

Reply:
Yep, we only track the sharing information if the logger is `Some`. That's currently the only case when the information is needed. Do you think we should always track this information?

Reply:
It would be more future-proof that way, but I'm fine with leaving it as is.

Out of scope for this PR, but I wonder if we could get rid of the `if let Some(logger)` pattern throughout the compute code and make the logger non-optional instead. I _think_ we always have a compute logger, except perhaps during initialization.